pkg/odo/cli/service/list.go

```diff
@@ -39,23 +42,57 @@ func NewServiceListOptions() *ServiceListOptions {
 
 // Complete completes ServiceListOptions after they've been created
 func (o *ServiceListOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.Context = genericclioptions.NewContext(cmd)
+	if experimental.IsExperimentalModeEnabled() {
+		o.Context = genericclioptions.NewDevfileContext(cmd)
+	} else {
+		o.Context = genericclioptions.NewContext(cmd)
+	}
 	return
 }
 
 // Validate validates the ServiceListOptions based on completed values
 func (o *ServiceListOptions) Validate() (err error) {
-	// Throw error if project and application values are not available.
-	// This will most likely be the case when user does odo service list from outside a component directory and
-	// doesn't provide --app and/or --project flags
-	if o.Context.Project == "" || o.Context.Application == "" {
-		return odoutil.ThrowContextError()
+	if !experimental.IsExperimentalModeEnabled() {
+		// Throw error if project and application values are not available.
+		// This will most likely be the case when user does odo service list from outside a component directory and
+		// doesn't provide --app and/or --project flags
+		if o.Context.Project == "" || o.Context.Application == "" {
+			return odoutil.ThrowContextError()
+		}
 	}
 	return
 }
 
 // Run contains the logic for the odo service list command
 func (o *ServiceListOptions) Run() (err error) {
+	if experimental.IsExperimentalModeEnabled() {
+		// if experimental mode is enabled, we list only operator hub backed
```

Comment:
what if we are running `openshift` with both? shouldn't we consider such a scenario as well?

Reply:
I don't get your question. Are you saying why not list the services started from service catalog as well in experimental mode? If that's the case, it's because a component in experimental mode doesn't have `app` value. 

As a result, at the moment, I've kept `odo service list` lose such that `odo service list` can be executed from anywhere on the system and it will list services started from operators. But for service catalog based services, you need an `app`. So I had to skip listing those.